VoodooI2CHID/VoodooI2CHIDDevice.cpp

```diff
@@ -38,27 +39,65 @@ bool VoodooI2CHIDDevice::init(OSDictionary* properties) {
         OSSafeReleaseNULL(clients);
         return false;
     }
+    
+    buf_i2c_cnt_intr = 0;
+    buf_i2c_cnt = 0;
+    buf_i2c_pool_intr = (UInt8 *)IOMalloc(I2C_MAX_BUF_SIZE);
+    buf_i2c_pool = (UInt8 *)IOMalloc(I2C_MAX_BUF_SIZE);
 
     return true;
 }
 
 void VoodooI2CHIDDevice::free() {
     if (client_lock)
         IOLockFree(client_lock);
+    
+    IOFree(buf_i2c_pool_intr, I2C_MAX_BUF_SIZE);
+    IOFree(buf_i2c_pool, I2C_MAX_BUF_SIZE);
+    if (read_in_progress_mutex) {
+        IOLockFree(read_in_progress_mutex);
+        read_in_progress_mutex = NULL;
+    }
 
     super::free();
 }
 
+UInt8* VoodooI2CHIDDevice::getMallocI2CIntr(UInt16 size) {
```

Comment:
What happens when the buffer is full? It seems like you overwrite memory.

Can you explain to me why we even need a buffer?